boa/src/builtins/object/mod.rs

```diff
@@ -877,3 +913,48 @@ fn object_define_properties(
     // 7. Return O.
     Ok(())
 }
+
+/// Type enum used in the abstract operation GetOwnPropertyKeys
+#[derive(Debug, Copy, Clone)]
+enum GetOwnPropertyKeysType {
```

Comment:
```suggestion
enum PropertyKeyType {
```
I don't think we need to prefix the type with the full function name, a simpler name should suffice.